app/models/registrant_user.rb

```diff
@@ -20,16 +20,34 @@ def country
   def companies(company_register = CompanyRegister::Client.new)
     return [] if ident.include?('-')
 
-    companies = company_register.representation_rights(citizen_personal_code: ident,
-                                                       citizen_country_code: country.alpha3)
-
-    companies = update_contacts_before_receive(companies)
-    companies
+    company_register.representation_rights(citizen_personal_code: ident,
+                                           citizen_country_code: country.alpha3)
   rescue CompanyRegister::NotAvailableError
-    return []
+    []
+  end
+
+  def do_need_update_contact?
+    return { result: false, counter: 0 } if companies.blank?
+
+    counter = 0
+    companies.each do |company|
+      contacts = Contact.where(ident: company.registration_number, ident_country_code: 'EE')
```

Comment:
its better to use one Active Record query here, than iterations:
`counter  += Contact.where(ident: company.registration_number, ident_country_code: 'EE')
                                   .where('name != ?', company.company_name).size`